mathesar/views/api.py

```diff
@@ -21,11 +22,18 @@ def create(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
 
-class TableViewSet(viewsets.ReadOnlyModelViewSet):
+class TableViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin):
     queryset = Table.objects.all().order_by('-created_at')
     serializer_class = TableSerializer
     pagination_class = DefaultLimitOffsetPagination
 
+    def create(self, request):
+        file_pk = request.data["file_pk"]
+        data_file = DataFile.objects.get(id=file_pk)
+        table = create_table_from_csv(data_file)
+        serializer = TableSerializer(table, context={'request': request})
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
```

Comment:
I suggest moving all logic out to some intermediary function that just takes the `request.data` blob, and then "does the right thing".  We'll want this `create` method to be quite generic and have the ability to handle multiple creation methods (not just from a CSV on the webserver disk).

Reply:
Agreed